### ISSUE TITLE
1.21 Trial Chambers are conspicuous in deep oceans

### ISSUE THREAD

**Issue by PepperCode1** [Apollounknowndev/tectonic#161]
Minecraft version: 1.21, vanilla
Tectonic version: 2.4.0a, unmodified
Other datapacks: none

Such plateaued, square terrain formations can be found across deep ocean biomes and are easily distinguishable from the otherwise flat ocean floor. There is always a trial chambers structure under them.

![image](https://github.com/Apollounknowndev/tectonic/assets/44146161/028ca689-e515-49fe-8f85-bedf95b5aa95)

**Comment by Rebel459** [comment]
imo the best solution here would be to disable their spawning in deep oceans. While having a separate structure set for them to gen them lower would typically be preferred, it is not possible to predict how much higher they will go, so removing them from those biomes entirely may be the best option

**Comment by Rebel459** [comment]
Though now that I think about it it would be challenging to do so without overriding the same tag from other mods

**Comment by Apollounknowndev** [comment]
> imo the best solution here would be to disable their spawning in deep oceans. While having a separate structure set for them to gen them lower would typically be preferred, it is not possible to predict how much higher they will go, so removing them from those biomes entirely may be the best option

Overriding the tag is dangerous, as if it loads above biome packs that add the trial chambers to their biomes the tag file gets overwritten. Overriding the structure file isn't great either though so idk.

**Comment by Rebel459** [comment]
Yes that is absolutely a concern, however I have noticed a couple things that could mitigate this. Many mods have seemingly forgotten to tag their biomes in trial chambers, simply because it does not use tags. A more creative solution could be to have the tag be part of a standalone Tectonic-mod datapack (like some of the other configs) which is enabled by default and completely overrides the tag to both remove the deep oceans _and_ include the other vanilla and modded biomes via using their tags rather than their direct IDs (eg the forest tag rather than listing each forest separately)

Doing this could both mitigate the chance of any biome tagging issues with other mods and potentially even increase the range of biomes the Trial Chambers generate in by default (obv niche cases where biomes are not tagged such as certain cave biomes would exist, but manual IDs could still be added and a config option would allow this to be disabled in a worst-case scenario)

In fact, I was thinking of making a datapack simply to make the Trial Chamber biome gen use tags for the sake of better compatibility with untagged mods, and a similar process could be applied here to maintain compatibility while having a fix for the ocean biomes, and any limitations could be mitigated with it being toggleable in the config.

**Comment by Rebel459** [comment]
Another option would be to try moving the trial chambers 10 blocks lower and see if it runs into any issues, however replacing biome ids with biome tag ids could be a good thing to look into

**Comment by Apollounknowndev** [comment]
Previously was holding out on modifying the structure file as Lithostitched did it as well but that is no longer the case. A future version will override the structure file.

**Comment by Apollounknowndev** [comment]
Fixed in 2.4.3.
